### PR TITLE
add --token-file option

### PR DESCRIPTION
**Pull request by jacalata**
You could previously put a token value in --password-file if you also passed in --token-name, but this is much clearer and makes it easier for users to discover and use the option.

### CODE REVIEW

File: tabcmd/commands/auth/session.py [hunk 1]
```diff
@@ -274,17 +276,11 @@ def create_session(self, args, logger):
...
+        if args.password or args.password_file:
             self._end_session()
             # we don't save the password anywhere, so we pass it along directly
             credentials = self._create_new_credential(args.password, Session.PASSWORD_CRED_TYPE)
-        elif args.password_file:
-            self._end_session()
-            if args.username:
-                credentials = self._create_new_credential(args.password, Session.PASSWORD_CRED_TYPE)
-            else:
-                credentials = self._create_new_credential(args.password, Session.TOKEN_CRED_TYPE)
```
Comment (anyoung-tableau) [inline tabcmd/commands/auth/session.py:286]: Looks like `Session.TOKEN_CRED_TYPE` is no longer ever passed to `_create_new_credential`. Can we remove its `credential_type` argument and rename the method to `_create_new_password_credential`?

File: tabcmd/commands/auth/session.py [hunk 2]
```diff
@@ -34,6 +34,7 @@ def __init__(self):
         self.token_name = None
         self.token_value = None
         self.password_file = None
+        self.token_file = None
```
Comment (anyoung-tableau) [inline tabcmd/commands/auth/session.py:37]: GitHub doesn't seem to allow me to leave comments on lines that haven't changed, but line 23-24 and 26-27 look duplicated. Is that some Python magic I'm not familiar with?

```python
    TOKEN_CRED_TYPE = "token"
    PASSWORD_CRED_TYPE = "password"

    TOKEN_CRED_TYPE = "token"
    PASSWORD_CRED_TYPE = "password"
```

Comment (jacalata) [reply tabcmd/commands/auth/session.py:37]: nope, that's just unnecessary duplication. Fixed.

File: tabcmd/commands/auth/session.py (hunk 1 above)
Comment (jacalata) [reply tabcmd/commands/auth/session.py:286]: yes - done

**Pull request merged**